esptool.py

```diff
@@ -281,6 +281,53 @@ def sync(self):
         for i in range(7):
             self.command()
 
+    def _connect_attempt(self, mode='default_reset', legacy=False):
+        """ A single connection attempt, with legacy handling """
+        last_error = None
+        # issue reset-to-bootloader:
+        # RTS = either CH_PD/EN or nRESET (both active low = chip in reset
+        # DTR = GPIO0 (active low = boot to flasher)
+        #
+        # DTR & RTS are active low signals,
+        # ie True = pin @ 0V, False = pin @ VCC.
+        if mode != 'no_reset':
+            self._port.setDTR(False)  # IO0=HIGH
+            self._port.setRTS(True)   # EN=LOW, chip in reset
+            if mode == 'esp32r0' or legacy:
+                time.sleep(2.0)  # Need to sleep longer here!
+            self._port.setDTR(True)   # IO0=LOW
+            self._port.setRTS(False)  # EN=HIGH, chip out of reset
+            if mode == 'esp32r0' or legacy:
```

Comment:
Also, for older boards `esp32r0` mode as an option only saves a fraction of a second (the default attempt is very quick). Given all the info I have now, it would be more maintainable to eliminate this named mode.